org-ql.el

```diff
@@ -433,7 +433,11 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                                      ;; Top-level heading: use file tags.
                                      org-file-tags)))
                                'org-ql-nil))
-           (all-tags (list inherited-tags local-tags)))
```

Comment:
I think this might be written more simply, preserving the let-binding structure, something like this:

```el
(let (...
      (all-tags (if org-group-tags
                    (list (org-ql--expand-tag-groups inherited-tags)
                          (org-ql--expand-tag-groups local-tags))
                  (list inherited-tags local-tags))))
  ...)
```

Reply:
That sounds reasonable to me too but I need to clarify something first. The values of both `local-tags` and `inherited-tags` should be modified when `org-group-tags` is non-nil to account for the groups, not just the value of `all-tags`. Do you suggest that it would be better to call `org-ql--expand-tag-groups` at the let bindings of `local-tags` and `inherited-tags` instead of updating them inside the body? I don't mind trying that but it seemed clearer to me. Let me know if that is what you prefer.

Reply:
I think you were right in the first place.  I hadn't looked at this code for a long time and didn't study it carefully enough before suggesting this.  :)  Feel free to change it back to the way it was when you first made the PR.

Reply:
To be clear, I mean to continue with testing for `org-ql-nil` here, but otherwise as before...if that makes sense...  :)

Reply:
No problem. I think I get what you mean. :)